Makefile

```diff
@@ -243,3 +243,12 @@ stubs:
 update-frozen-libraries:
 	@echo "Updating all frozen libraries to latest tagged version."
 	cd frozen; for library in *; do cd $$library; ../../tools/git-checkout-latest-tag.sh; cd ..; done
+
+one-of-each: all-source
+	make -C ports/atmel-samd BOARD=trinket_m0
+	make -C ports/atmel-samd BOARD=feather_m4_express
+	make -C ports/esp32s2 BOARD=espressif_saola_1_wroom
+	make -C ports/litex BOARD=fomu
+	make -C ports/mimxrt10xx BOARD=feather_mimxrt1011
+	make -C ports/nrf BOARD=feather_nrf52840_express
+	make -C ports/stm BOARD=feather_stm32f405_express
```

Comment:
I've had something like this locally for some time.  

It's more verbose in the Makefile, but if you write it like this, you can specify parallelism (e.g., `make -j8` if you have an 8-core CPU) and get faster builds.  However, as the output of various commands is interleaved, it can also make failures harder to interpret.  On the whole, I like faster:
```
one-of-each: one-of-each-1
one-of-each-1: all-source
        $(MAKE) -C ports/atmel-samd BOARD=trinket_m0
one-of-each: one-of-each-2
one-of-each-2: all-source
        $(MAKE) -C ports/atmel-samd BOARD=feather_m4_express
one-of-each: one-of-each-3
one-of-each-3: all-source
        $(MAKE) -C ports/esp32s2 BOARD=espressif_saola_1_wroom
one-of-each: one-of-each-4
one-of-each-4: all-source
        $(MAKE) -C ports/litex BOARD=fomu
one-of-each: one-of-each-5
one-of-each-5: all-source
        $(MAKE) -C ports/mimxrt10xx BOARD=feather_mimxrt1011
one-of-each: one-of-each-6
one-of-each-6: all-source
        $(MAKE) -C ports/nrf BOARD=feather_nrf52840_express
one-of-each: one-of-each-7
one-of-each-7: all-source
```